core/api_client.py

```
@@ -1,35 +1,59 @@
         load_dotenv()
-        api_key = os.getenv("ANTHROPIC_API_KEY")
-        if not api_key:
-            raise ValueError("ANTHROPIC_API_KEY not found in environment variables")
-        self.client = Anthropic(api_key=api_key)
-        self.model = "claude-3-5-sonnet-20240620"
+        self.provider = provider
+        
+        if self.provider == "anthropic":
+            api_key = os.getenv("ANTHROPIC_API_KEY")
+            if not api_key:
+                raise ValueError("ANTHROPIC_API_KEY not found in environment variables")
```

Ruff (0.14.8)

Avoid specifying long messages outside the exception class

(TRY003)